Fix supportsArrayTranslate and supportsArraySet

knn-k opened the pull request:
This commit fixes the condition for supporting arraytranslate and arrayset operations. They work only with contiguous arrays.

knn-k commented:
Jenkins build plinux,amac

0xdaryl commented:
Your commit also affects ArraySet, which isn't an array translation.  Can you amend your commit title and message to indicate that?

That's assuming you did that intentionally.  I believe it is true that arrayset (on either P or AArch64) does not handle arraylets.

knn-k commented:
Jenkins build plinux,amac

knn-k commented:
I updated the commit title and the commit message as suggested.

AArch64 arraysetEvaluator does not take arraylets into consideration.
https://github.com/eclipse-omr/omr/blob/69377777255ae91ec11ec373574a2718f07610cd/compiler/aarch64/codegen/OMRTreeEvaluator.cpp#L6560

P arrayset optimization has been disabled by default, and it does not care about arraylets, either.  I moved its `setSupportsArraySet()` into the `if (!TR::Compiler->om.canGenerateArraylets()) { ... }` block.
https://github.com/eclipse-omr/omr/blob/69377777255ae91ec11ec373574a2718f07610cd/compiler/p/codegen/TreeEvaluatorVMX.cpp#L53

knn-k commented:
I have run sanity.functional, sanity.system, and sanity.openjdk in https://github.com/eclipse-openj9/openj9/pull/20867 by depending on this PR.  They were all successful.
